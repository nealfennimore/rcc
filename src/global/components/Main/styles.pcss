.Main {
    grid-area: main;
    padding: 30px;

    &.center {
        display: flex;
        align-items: center;
        justify-content: center;
    }
}
