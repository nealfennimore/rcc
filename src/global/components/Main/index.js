import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './styles.pcss';

const Main = ( { children, center } ) => (
    <main className={classnames( styles.Main, {
        [styles.center]: center,
    } )}
    >
        { children }
    </main>
);

Main.propTypes = {
    children: PropTypes.node,
    center: PropTypes.bool,
};

export default Main;
