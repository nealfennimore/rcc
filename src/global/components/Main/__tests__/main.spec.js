import React from 'react';
import renderer from 'react-test-renderer';
import Main from '../index.js';

describe( 'Main', () => {
    it( 'should match snapshot', () => {
        const component = renderer.create(
            <Main />,
        );
        expect( component.toJSON() ).toMatchSnapshot();
    } );
    it( 'should match snapshot when centered', () => {
        const component = renderer.create(
            <Main center />,
        );
        expect( component.toJSON() ).toMatchSnapshot();
    } );
} );
