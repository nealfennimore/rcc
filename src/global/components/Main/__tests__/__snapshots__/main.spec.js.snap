// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
<main
  className="Main"
/>
`;

exports[`Main should match snapshot when centered 1`] = `
<main
  className="Main center"
/>
`;
